source/sio/io_uring/async_accept.hpp

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright (c) 2023 Xiaoming Zhang
+ *
+ * Licensed under the Apache License Version 2.0 with LLVM Exceptions
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *   https://llvm.org/LICENSE.txt
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include "sio/concepts.hpp"
+#include "sio/sequence/sequence_concepts.hpp"
+#include "sio/io_uring/file_handle.hpp"
+#include "sio/net/ip/address.hpp"
+
+#include <exec/env.hpp>
+#include <exec/linux/io_uring_context.hpp>
+#include <exec/trampoline_scheduler.hpp>
+#include <sys/socket.h>
+
+namespace sio {
+  namespace accept_ {
+    using namespace stdexec;
+
+    // TODO: with async_allocator
+    // using sender_t = async_allocator(sender_t);
+    using sender_t = io_uring::accept_sender;
+
+    template <class Receiver>
+    struct operation;
+
+    template <class Receiver>
+    struct next_receiver {
+      using is_receiver = void;
+
+      operation<Receiver>* op_;
+
+      void set_value(set_value_t) && noexcept {
+        op_->start_next();
```

Comment:
If you simply reuse the operation state after the previous next-sender-op completes you will only have one active connection at the time. Instead we need logic where a new operation state is created and started when the accept-sender completes and when the next-sender completes we can discard this operation state. This is where we need dynamic allocation.